src/generators/MongoidTypeGenerator.ts

```diff
@@ -0,0 +1,31 @@
+import FileWriter from "../FileWriter";
+import { mongoidPrimitiveTypes } from "../model/dataTypes/primitiveDataTypes";
+import TypeInfo from "../model/TypeInfo";
+import classTemplate from "../templates/rubymongoid/classTemplate";
+import IGenerator from "./IGenerator";
+
+export default class MongoidTypeGenerator implements IGenerator {
```

Comment:
@AndrewBird81 I just copy pasted this from typescript model generator. we can discuss if we want to introduce abstract class or move some common code

Reply:
yeah, go ahead and make an abstract class